Dockerfile.ubi

```
@@ -0,0 +1,232 @@
+ENV LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8 \
+    UV_LINK_MODE=copy \
+    TORCH_CUDA_ARCH_LIST="9.0;10.0+PTX" \
+    PYTHON_VERSION=${PYTHON_VERSION} \
+    UV_TORCH_BACKEND=${UV_TORCH_BACKEND:-cu128} \
+    VIRTUAL_ENV=/opt/vllm 
+
+# Install base packages and EPEL in single layer
+RUN dnf install -y dnf-plugins-core && \
+    dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    dnf config-manager --set-enabled epel && \
+    dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo && \
+    dnf install -y --allowerasing \
+        python${PYTHON_VERSION} python${PYTHON_VERSION}-pip python${PYTHON_VERSION}-wheel \
+        python${PYTHON_VERSION}-devel \
+        # use nvidia's packaged nvshmem
+        nvshmem-cuda-12 \ 
+        which procps findutils tar \
+        gcc gcc-c++ \
+        make cmake \
+        autoconf automake libtool \
+        git \
+        curl wget \
+        gzip \
+        zlib-devel \
+        openssl-devel \
+        pkg-config \
+        libuuid-devel \
+        glibc-devel \
+        rdma-core-devel \
+        numactl-libs \
+        subunit \
+        pciutils \
+        pciutils-libs \
+        ninja-build \
+    && dnf clean all
+
+# Setup Python virtual environment
+RUN python${PYTHON_VERSION} -m venv /opt/vllm && \
+    ${VIRTUAL_ENV}/bin/pip install --no-cache -U pip wheel uv meson-python ninja pybind11
+
+ENV LIBRARY_PATH="/usr/local/cuda/lib64:/usr/local/lib:/usr/local/lib64" \
+CPATH="/usr/include:/usr/local/include:/usr/local/cuda/include" \
+PKG_CONFIG_PATH="/usr/lib/pkgconfig:/usr/local/lib/pkgconfig:/usr/local/lib64/pkgconfig"
+
+# Set NVSHMEM paths for CMake discovery
+ENV NVSHMEM_DIR=/usr/lib64/nvshmem/12 \
+    PATH="/usr/bin/nvshmem_12:${PATH}" \
+    CPATH="/usr/include/nvshmem_12:${CPATH}" \
+    LIBRARY_PATH="/usr/lib64/nvshmem/12:${LIBRARY_PATH}"
+
+# Build and install gdrcopy
+RUN --mount=type=cache,target=/var/cache/git \
+    git clone https://github.com/NVIDIA/gdrcopy.git && \
+    cd gdrcopy && \
+    PREFIX=/usr/local DESTLIB=/usr/local/lib make lib_install && \
+    cp src/libgdrapi.so.2.* /usr/lib64/ && \
+    ldconfig && \
+    cd .. && rm -rf gdrcopy
+
+# Build and install UCX
+RUN --mount=type=cache,target=/var/cache/git \
+    git clone https://github.com/openucx/ucx.git && \
+    cd ucx && \
+    git checkout v1.18.0 && \
+    ./autogen.sh && \
+    ./configure \
+        --enable-shared \
+        --disable-static \
+        --disable-doxygen-doc \
+        --enable-cma \
+        --enable-devel-headers \
+        --with-cuda=/usr/local/cuda \
+        --with-verbs \
+        --with-dm \
+        --with-gdrcopy=/usr/local \
+        --enable-mt \
+        --with-mlx5 \
+        --prefix=/usr/local && \
+    make -j$(nproc) && \
+    make install && \
+    ldconfig && \
+    cd .. && rm -rf ucx
+
+ARG NIXL_REPO_URL="https://github.com/ai-dynamo/nixl.git"
+ARG NIXL_COMMIT_SHA="v0.3.0"
```

Looks like it should be this:
```suggestion
ARG NIXL_COMMIT_SHA="0.3.0"
```

Should we upgrade to 0.4.0 or is there some specific commit we should be grabbing (@robertgshaw2-redhat)?

---

I think UCX 1.18.1 is fine for this release. Talking to AWS folks, newer versions are needed for their networking